service/src/main/java/bio/terra/workspace/db/WorkspaceDao.java

```diff
@@ -520,27 +662,66 @@ public void updateWorkspaceProperties(UUID workspaceUuid, Map<String, String> pr
    * @param idList List of workspaceIds to query for
    * @param offset The number of items to skip before starting to collect the result set.
    * @param limit The maximum number of items to return.
-   * @return list of Workspaces corresponding to input IDs.
+   * @return map of Workspaces descriptions indexed by workspace id
    */
   @Traced
   @ReadTransaction
-  public List<Workspace> getWorkspacesMatchingList(Set<UUID> idList, int offset, int limit) {
+  public Map<UUID, DbWorkspaceDescription> getWorkspacesMatchingList(
```

Comment:
getWorkspaceIdToDescriptionMap

Reply:
I made it `getWorkspaceDescriptionMapFromIdList`